test/fixtures/parser/bigquery/select_udf_quote_nothing.yml

```diff
@@ -0,0 +1,33 @@
+file:
+  statement:
+    select_statement:
+      select_clause:
+        keyword: SELECT
+        select_clause_element:
+          function:
+          - identifier: gcpproject
+          - dot: .
+          - identifier: functions
+          - dot: .
+          - function_name:
+              function_name_identifier: timestamp_parsing
```

Comment:
Hmmmmm, I wonder whether the `identifiers` should be _part of_ `function_name`, i.e. that it should parse as:

```
function:
- function_name:
  - identifier: gcpproject
  - dot: .
  - idenfifier: functions
  - dot: .
  - function_name_identifier: timestamp_parsing
- start_bracket: (
...
```

Otherwise it feels wierd to nest `function_name_identifier` within `function_name` if there's never anything in there with it.

Reply:
I'll look into this after I make the simpler changes. This PR had involved some "trial and error" and did kind of end up in a weird place. The main challenge, as I recall, was to allow hyphens in function names at the parse level, since it apparently couldn't be done at the lexer level.

Reply:
Never mind about the hyphens -- I'm confusing this with an earlier PR for project names in the `FROM` clause, which *did* allow non-quoted hyphens. That one was a little odd. I think a more straightforward approach should work here.